framework/sandstone.cpp

```diff
@@ -2592,6 +2592,8 @@ static struct test *get_next_test(int tc)
     }
 
     auto next_test = test_selector->get_next_test();
+    while (next_test && next_test->quality_level < sApp->requested_quality)
```

Comment:
We already have this code somewhere...

Reply:
Apparently we don't. Otherwise #213 would not have existed. We have similar code in test listings...